org.adempiere.base/src/org/compiere/model/MLookup.java

```diff
@@ -995,6 +996,86 @@ private synchronized static CCache<String, ValueNamePair> getDirectValueNamePair
 		return vnpCache;
 	}
 	
+	/**
+	 * Get Lookup
+	 * @param tableID
+	 * @param windowNo
+	 * @param tabNo
+	 * @return null if tableID <= 0 or the table doesn't have any key column, else {@link MLookup}
+	 */
+	public static MLookup getRecordsLookup(int tableID, int windowNo, int tabNo) {
+		return getRecordsLookup(tableID, windowNo, tabNo, null);
+	}
+	
+	/**
+	 * Get Lookup
+	 * @param tableID
+	 * @param windowNo
+	 * @param tabNo
+	 * @param keyColumn - override the default key column for the lookup

```

Comment:
In practise, this is either using the table's single key column or using uuid column. In otherword, you just need a boolean flag, say "useUUIDKey" to control that. With this change, I think you don't need the getKeyColumn(table) method at WRecordIDEditor either.

Reply:
Ok, I've replaced the keyColumn parameter with a boolean flag, but I think instead of the `getKeyColumn(table) `method we still should use at least a `boolean isUseUUIDKey()` abstract method, because most of the calls of the getIdentifier methods are in the abstract WRecordEditor class. All changes committed.